test/Microsoft.ML.TimeSeries.Tests/TimeSeriesDirectApi.cs

```diff
@@ -571,6 +575,93 @@ public void AnomalyDetectionWithSrCnn(bool loadDataFromFile)
             }
         }
 
+        [Theory, CombinatorialData]
+        public void TestSrCnnBatchAnomalyDetector(
+            [CombinatorialValues(SrCnnDetectMode.AnomalyOnly, SrCnnDetectMode.AnomalyAndExpectedValue, SrCnnDetectMode.AnomalyAndMargin)]SrCnnDetectMode mode,
+            [CombinatorialValues(true, false)]bool loadDataFromFile)
+        {
+            var ml = new MLContext(1);
+            IDataView dataView;
+            if (loadDataFromFile)
+            {
+                var dataPath = GetDataPath(Path.Combine("Timeseries", "anomaly_detection.csv"));
+
+                // Load data from file into the dataView
+                dataView = ml.Data.LoadFromTextFile(dataPath, new[] {
```

Comment:
>LoadFromTextFile [](start = 35, length = 16)

`LoadFromTextFile` has a generic overload that you can use here with `TimeSeriesDataDouble` instead of specifying the `TextLoader.Column`.

Reply:
I've tried dataView = ml.Data.LoadFromTextFile<TimeSeriesDataDouble>(dataPath, hasHeader: true); But the test failed due to System.InvalidOperationException : Property 'Value' is missing the LoadColumnAttribute attribute

---
In reply to: [426442626](https://github.com/dotnet/machinelearning/pull/5135#discussion_r426442626) [](ancestors = 426442626)

Reply:
Fixed